addons/web/tests/test_js.py

```diff
@@ -11,8 +11,29 @@
 class WebSuite(odoo.tests.HttpCase):
 
     def test_js(self):
+        # this is just a poc only working if positive tags are before negative tags
+        # usually only negative or positive tags should be used together
+        positive = []
+        negative = []
+        for param in self._test_params:
+            for filter in param.split(','):
+                if not filter:
+                    continue
+                if filter.startswith('-'):
+                    filter = filter[1:]
+                    negative.append(f'(?!.*{filter})')
+                else:
+                    positive.append(f'(.*{filter})')
+
+        filter_param = ''
+        if positive or negative:
+            positive = '|'.join(positive)
+            negative = ''.join(negative)
+            filter = f'^({positive}){negative}.*$'
+            filter_param = f'&filter=/{filter}/'
+
         # webclient desktop test suite
-        self.phantom_js('/web/tests?mod=web&failfast', "", "", login='admin', timeout=1800)
+        self.phantom_js('/web/tests?mod=web&failfast%s' % filter_param, "", "", login='admin', timeout=1800)
```

Comment:
At this point, this needs to be urlencoded, the querystring is getting really complicated and potentially broken.